### ISSUE TITLE
Unable to find u_WECstress

### ISSUE THREAD

**Issue by able93** [jcwarner-usgs/COAWST#266]
Hello,

I am trying to restart my coupled ROMS-SWAN model. However, it gives the error unable to find the variable u_WECstress in history file. Meanwhile, I have the variable in the history file. What could have caused that?

Thanks

**Comment by jcwarner-usgs** [comment]
if you deactivate u_WECstress from being written to the his file, does the error go away?
i have not seen that one before.  can you show the full out?

**Comment by able93** [comment]
I have attached the log file.

I tried your first suggestion of not writing out the u_WECstress, and got this error:

     14140 2011-01-15 17:30:00.00  2.513840E-02  1.315028E+04  1.315031E+04  1.397539E+15
                     (558,500,32)  3.996223E-02  1.084959E-01  0.000000E+00  2.559730E+00
  DEF_HIS_NF90     - inquiring history file,           Grid 01: ./output/veg_fb_his_1_00354.nc
  DEF_AVG_NF90     - inquiring average file,           Grid 01: ./output/veg_fb_avg_1_00014.nc
  DEF_RST_NF90     - inquiring restart file,           Grid 01: ./output/veg_fb_rst_1.nc
At line 828 of file def_rst.f90 (unit = 6, file = 'stdout')
Fortran runtime error: Expected REAL for item 1 in formatted transfer, got CHARACTER
(1pe11.4,1x,'millimeter')
   ^

Error termination. Backtrace:
+time 20110115.183500   , step      1; iteration    1; sweep 3 grid   1
#0  0x7f44343b5319 in require_type
	at ../../../libgfortran/io/transfer.c:1321
#1  0x7f44343b7de0 in require_type
	at ../../../libgfortran/io/transfer.c:1313

Thanks

[fbveg (1) - Copy.log](https://github.com/DOI-USGS/COAWST/files/15472136/fbveg.1.-.Copy.log)

**Comment by jcwarner-usgs** [comment]
not sure. what does this look like:
At line 828 of file def_rst.f90

**Comment by able93** [comment]
# ifdef PERFECT_RESTART
        status=def_var(ng, iNLM, RST(ng)%ncid, RST(ng)%Vid(idbath),     &
     &                 NF_FRST, nvd4, t2dgrd, Aval, Vinfo, ncname,      &
     &                 SetFillVal = .FALSE.)
# else
        status=def_var(ng, iNLM, RST(ng)%ncid, RST(ng)%Vid(idbath),     &
     &                 NF_FRST, nvd3, t2dgrd, Aval, Vinfo, ncname,      &
     &                 SetFillVal = .FALSE.)
# endif
        IF (FoundError(exit_flag, NoError, __LINE__, MyFile)) RETURN
#endif

meanwhile, I didn't define perfect restart

**Comment by able93** [comment]
So the error was due to incomplete filepath for the restart file. It now starts properly. But now, when it is time to write average file, it stops again with below error.

Found Error: **   Line: 700      Source: ROMS/Utility/wrt_avg.F, wrt_avg_nf90

 WRT_AVG_NF90 - error while writing variable: Lwave
                into averages NetCDF file for time record: 2
 Found Error: 03   Line: 96       Source: ROMS/Utility/wrt_avg.F
 Found Error: 03   Line: 414      Source: ROMS/Nonlinear/output.F
 Found Error: 03   Line: 573      Source: ROMS/Nonlinear/main3d.F
 Found Error: 03   Line: 330      Source: ROMS/Drivers/nl_roms.h, ROMS_run

**Comment by jcwarner-usgs** [comment]
did it write any other values to the avg file?
did it have Lwave for time step 1?

**Comment by able93** [comment]
Yes, it did write some other values, but stopped at the Lwave. It does this when I restart and it needs to write the average file. I am trying to reproduce the process to see exactly why it does that. I discover it does this whenever I use a restart file to start the model. I was also thinking maybe because it sees an old average file and cannot write into it. Then I tried restarting with a history file and it works, even though I deleted the old average file before restarting. 

Meanwhile, I created a vegetation density file, and put in the swan file for it to read, but it gives error while reading the file, and just prints all the values in the file into PRINT-001. 
&& KEYWORD TO CREATE PLANTS &&
INPGRID NPLANTS CURVILINEAR 0 0 641 561 EXC 9.999000e+003 &
       NONSTATIONARY 20110101.000000 3 HR 20121231.000000
READINP NPLANTS 1 '/mnt/beegfs/swan_veg_parent.dat' 4 0 FREE  

VEGETATION 1 0.3 0.23 300 1.0 

[swan_veg_parent.txt](https://github.com/user-attachments/files/15517305/swan_veg_parent.txt)


What could have caused that?

Thanks
